src/Columns/ColumnLazy.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <cstring>
+#include <cassert>
+
+#include <Columns/IColumn.h>
+#include <Columns/IColumnImpl.h>
+#include <Common/PODArray.h>
+#include <Common/assert_cast.h>
+#include <Core/Field.h>
+
+
+namespace DB
+{
+
+/** Column for lazy materialization.
+  */
+class ColumnLazy final : public COWHelper<IColumn, ColumnLazy>
+{
+private:
+    friend class COWHelper<IColumn, ColumnLazy>;
+
+    WrappedPtr part_nums;
```

Comment:
I thought it would store a callback that encapsulates all the details, and you initialize the column with this callback. It will be useful for any kind of delayed reading or calculations.

Reply:
If I understand correctly, I need to encapsulate row_nums and part_nums into an array. Different delayed executions may require different column arrays. It is similar to ColumnFunction.